src/components/WateringModal/index.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { FC, useState } from 'react';
+import ButtonRound from '../ButtonRound';
+import { NumberInput } from '../NumberInput';
+import styled from 'styled-components';
+import { useWateringActions } from '../../utils/hooks/useWateringActions';
+import { useCurrentTreeId } from '../../utils/hooks/useCurrentTreeId';
+import { Modal } from '../Modal';
+import { DatePickerDialog } from '../DatePickerDialog';
+
+const TwoColumnGrid = styled.div`
+  width: 100%;
+  display: grid;
+  grid-template-columns: 1fr;
+  gap: 8px;
+
+  @media (min-width: ${p => p.theme.screenWidthS}) {
+    grid-template-columns: repeat(2, 1fr);
+  }
+`;
+
+const ButtonsContainer = styled.div`
+  margin-top: 48px;
+  display: flex;
+  justify-content: space-between;
+`;
+
+const StyledError = styled.p`
+  grid-column: 1 / 3;
+  color: ${p => p.theme.colorAlarm};
+  margin: 0;
+`;
+
+export const WateringModal: FC<{
+  isOpen?: boolean;
+  setIsOpen?: (isOpen: boolean) => void;
+}> = ({ isOpen = false, setIsOpen = () => undefined, children }) => {
+  const treeId = useCurrentTreeId();
+  const { waterTree, isBeingWatered } = useWateringActions(treeId);
+
+  const [wateringValue, setWateringValue] = useState(0);
+  const [wateringDate, setWateringDate] = useState(new Date());
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  const handleWatering = () => {
+    if (isBeingWatered) return;
+    if (wateringValue === 0) {
+      setError('Bitte gib an, wie viele Liter Du gegossen hast.');
+      return;
+    }
+    if (wateringValue >= 1000) {
+      setError('Eine so große Bewässerung kann nicht eingetragen werden.');
+      return;
+    }
+
+    // NOTE: We force the watering date to be in the afternoon instead of at 00:00:00 to avoid involutary date changes:
+    const wateringDateAfternoon = new Date(wateringDate.setHours(15));
+
+    void waterTree(wateringValue, wateringDateAfternoon).finally(() => {
```

Comment:
As I understand it, `.finally()` fires event when the promise fails. But here you call `setError(undefined)`. Shouldn't there be some kind of error handling here? Plz ignore if I just misunderstood.

Reply:
Fixed! Thanks for the discussion off-GitHub!